src/mainwindow.cpp

```diff
@@ -8320,29 +8320,22 @@ QTreeWidgetItem *MainWindow::addTagToTagTreeWidget(QTreeWidgetItem *parent,
                         continue;
                     }
 
-                    linkedNoteIds << Tag::fetchAllLinkedNoteIdsForFolder(
-                                         tagIdToCount,
-                                         folder, showNotesFromAllSubFolders,
-                                         isShowNotesRecursively);
+                    count = Tag::countLinkedNoteFileNamesForNoteSubFolder(
+                                tagIdToCount, folder,
+                                showNotesFromAllSubFolders, isShowNotesRecursively);
                 }
             }
         } else {
-            linkedNoteIds.reserve(tagIdListToCount.size());
             for (const int tagToCount : tagIdListToCount) {
-                linkedNoteIds << Tag::fetchAllLinkedNoteIds(
+                count =  Tag::countLinkedNoteFileNames(
```

Comment:
@Waqar144, this doesn't work if `tagIdListToCount` contain multiple tags (when `Tag::isTaggingShowNotesRecursively()` is true), because it will overwrite the note count all the time. This will cause #2819.

Reply:
indeed, fortunately easy to reverse this particular change.

Reply:
I already tried in [qownnotes.patch.gz](https://github.com/pbek/QOwnNotes/files/12466079/qownnotes.patch.gz), but it isn't working properly yet...

Reply:
Full revert will be difficult, it will be easier to just revert this portion `count = ...Tag::countAll..` to `linkedNoteIds << Tag:::countAll..`. It's too old to be fully reverted and we changed apis a bit

Reply:
So basically it will be a fix, not really a revert.

Reply:
Yes, that's what I tried in above patch, but it didn't work yet...